Transport.php

```diff
@@ -72,4 +80,73 @@ public static function exists($transport)
     {
         return in_array(strtolower($transport), self::get());
     }
+
+    /**
+     * Register a new wrapper in the collection
+     *
+     * @param  string   $protocol The registered protocol ws, wss, http...
+     * @param  callable $factory  The factory callable
+     * @return void
+     */
+    public static function registerWrapper($protocol, callable $factory)
+    {
+        if( isset(static::$_wrappers[$protocol]) ) {
+            throw new Exception(
+                'The protocol wrapper for %s is already registered!',
+                0,
+                $protocol);
+        }
+
+        static::$_wrappers[$protocol] = $factory;
+
+        return;
+    }
+
+    /**
+     * Share a wrapper factory between different protocols
+     * @param  string $protocol The protocol to used
+     * @param  string $shared   The shared used as factory
+     * @return void
+     */
+    public static function shareWrapper($protocol, $shared)
```

Comment:
It'll allow to share the same factory between different protocols. 

```php
<?php
Hoa\Socket\Transport::registerWrapper('ws', function() {...});
Hoa\Socket\Transport::shareWrapper('wss', 'ws);
```

I think it can be useful in our case...